src/Simulation/Native/src/simulator/wavefunction.hpp

```diff
@@ -238,19 +290,55 @@ class Wavefunction
         rng_.seed(s);
     }
 
+    void reorderWFN(unsigned qubitLoc, unsigned newPos) const
```

Comment:
in kernels.hpp there is:
`
template <class T, class A>
void swap(std::vector<T, A>& wfn, unsigned q1, unsigned q2)
`
how is it different from what you are trying to do here?

Reply:
I believe that is the swap gate operation. This is meant to be the reordering routine - which updates the mapping and the wavefunction. Also in it s final form, this reordering routine will not just 2 qubits to be swapped with each other but a collection of them.

Reply:
Also the abbreviation is in accordance with the rest of the file also using WFN

Reply:
Would it make sense to implement `swap` in terms of reorder_wavefunction?
Ss far as naming is concerned, having local variables named like "wfn" might be fine because the context is local and clear, but function names serve as documentation, especially at the callsites, and in talking about code we often refer to function names so it makes sense for them to be pronouncable...

Reply:
That could be an interesting proposition, but as it is not directly related to the purpose of this feature branch, I personally feel it could be more appropriate incorporated with a different PR appropriately linked to an issue.

Reply:
To your discretion. It might be confusing to readers of the code (as it was to me) to see two differently named functions that are implemented exactly the same, but it's not a big deal.